### PR TITLE
docs(react): iterator type on es5 target (#1900)

### PR DESCRIPTION
**Pull request by Himself65**
## Related Issues or Discussions

Fixes #1900

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: src/react/utils/useHydrateAtoms.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ type AnyWritableAtom = WritableAtom<unknown, any[], any>
 const hydratedMap: WeakMap<Store, WeakSet<AnyWritableAtom>> = new WeakMap()
 
 export function useHydrateAtoms(
-  values: Iterable<readonly [AnyWritableAtom, unknown]>,
+  values: (readonly [AnyWritableAtom, unknown])[],
```
Comment (dai-shi) [inline src/react/utils/useHydrateAtoms.ts:11]: Does this support `Map`?

Comment (Himself65) [reply src/react/utils/useHydrateAtoms.ts:11]: No. I didn't realize here accept a map

Comment (dai-shi) [reply src/react/utils/useHydrateAtoms.ts:11]: #1898 will be adding a test with `Map`.

Comment (Himself65) [reply src/react/utils/useHydrateAtoms.ts:11]: Ok i will update the document instead of code

**Review by dai-shi — APPROVED** [review]
Thanks!!

**Pull request merged**